core/primitives/src/views.rs

```diff
@@ -212,7 +212,8 @@ pub struct StateItem {
 #[derive(BorshSerialize, BorshDeserialize, Serialize, Deserialize, Debug, PartialEq, Eq, Clone)]
 pub struct ViewStateResult {
     pub values: Vec<StateItem>,
-    pub proof: TrieProofPath,
+    // signals whether the value was found in the trie, and the proof itself to verify the claim
+    pub proof: (bool, TrieProofPath),
```

Comment:
Isn’t it clear from `values` being empty that the value is not present in the trie?

Also, the correct type for this would be `Option<(bool, TrieProofPath)>`.  That is, there’s either no proof, of there is a proof of value being or not being there.

Reply:
`None` would be returned when the user requests no proof then?

Reply:
Right.  I still worry a little about who we might break if we change the type of the field so abruptly but yes the general idea would be `None` means no proof was requested and thus no proof was provided, `Some((found, proof))` means proof for existence / non-existence of the value is provided.